roles/variables/vars/main.yml

```diff
@@ -89,6 +89,176 @@ zookeeper_combined_properties: "{{ zookeeper_properties | confluent.platform.com
+      controller.quorum.voters: "{% for inventory_hostname in groups.kafka_controller|default([]) %}{% if loop.index > 1%},{% endif %}{{inventory_hostname[-1]|int + 9990}}@{{inventory_hostname}}:9093{%endfor%}"
+      controller.listener.names: "{{kafka_controller_listeners['controller']['name']}}"
+      listener.security.protocol.map: "{% for listener in kafka_controller_listeners|dict2items %}{% if loop.index > 1%},{% endif %}{{ listener['value']['name'] }}:{{ listener['value'] | confluent.platform.kafka_protocol_defaults(kafka_controller_ssl_enabled, kafka_controller_sasl_protocol)}}{% endfor %}"
+      listeners: "{{kafka_controller_listeners['controller']['name']}}://:{{kafka_controller_listeners['controller']['port']}}"
+      node.id: "{{ inventory_hostname[-1]|int + 9990 }}"
+  kraft_sasl_plain:
+    enabled: "{{kafka_controller_sasl_protocol=='plain'}}"
+    properties:
+      listener.name.controller.plain.sasl.jaas.config: "{{plain_jaas_config}}"
+      sasl.mechanism.controller.protocol: PLAIN
+  ssl:
+    enabled: "{{
+    kafka_controller_listeners | confluent.platform.ssl_required(ssl_enabled) or
+    kafka_controller_rest_ssl_enabled|bool or
+    mds_broker_listener.ssl_enabled|bool or
+    mds_tls_enabled|bool or
+    ( kafka_controller_schema_validation_enabled and 'schema_registry' in groups and schema_registry_ssl_enabled ) }}"
+    properties:
+      ssl.truststore.location: "{{kafka_controller_truststore_path}}"
+      ssl.truststore.password: "{{kafka_controller_truststore_storepass}}"
+      ssl.keystore.location: "{{kafka_controller_keystore_path}}"
+      ssl.keystore.password: "{{kafka_controller_keystore_storepass}}"
+      ssl.key.password: "{{kafka_controller_keystore_storepass}}"
+  fips:
+    enabled: "{{fips_enabled}}"
+    properties:
+      enable.fips: 'true'
+      security.providers: io.confluent.kafka.security.fips.provider.BcFipsProviderCreator,io.confluent.kafka.security.fips.provider.BcFipsJsseProviderCreator
+      ssl.keymanager.algorithm: PKIX
+      ssl.trustmanager.algorithm: PKIX
+      ssl.keystore.type: BCFKS
+      ssl.truststore.type: BCFKS
+      ssl.truststore.location: "{{kafka_controller_truststore_path}}"
+      ssl.truststore.password: "{{kafka_controller_truststore_storepass}}"
+      ssl.keystore.location: "{{kafka_controller_keystore_path}}"
+      ssl.keystore.password: "{{kafka_controller_keystore_storepass}}"
+      ssl.key.password: "{{kafka_controller_keystore_storepass}}"
+      ssl.enabled.protocols: TLSv1.2
+  sasl_enabled:
+    enabled: "{{ kafka_controller_sasl_enabled_mechanisms|length > 0 }}"
+    properties:
+      sasl.enabled.mechanisms: "{% for mechanism in kafka_controller_sasl_enabled_mechanisms %}{% if loop.index > 1%},{% endif %}{{ mechanism|upper }}{% endfor %}"
+  sasl_gssapi:
+    enabled: "{{ 'GSSAPI' in kafka_controller_sasl_enabled_mechanisms }}"
+    properties:
+      sasl.kerberos.service.name: "{{kerberos_kafka_controller_primary}}"
+  sr:
```

Comment:
SR related properties might not work in the controller.
Let's verify once and then we can remove them. Not urgent and we can have a task for this in place and fix that later.